clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -690,6 +691,112 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+
+            @Override
+            public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+                // set initial position so we don't need a lookup
+                for (TopicPartition partition : partitions)
+                    consumer.seek(partition, 0);
+            }
+        });
+
+        assertTrue(consumer.assignment().isEmpty());
+
+        client.prepareResponseFrom(new GroupCoordinatorResponse(Errors.NONE.code(), node).toStruct(), node);
+        Node coordinator = new Node(Integer.MAX_VALUE - node.id(), node.host(), node.port());
+        client.prepareResponseFrom(joinGroupFollowerResponse(assignor, 1, "memberId", "leaderId", Errors.NONE.code()), coordinator);
+        client.prepareResponseFrom(syncGroupResponse(Arrays.asList(partition1), Errors.NONE.code()), coordinator);
+
+        consumer.poll(0);
+        assertTrue(consumer.assignment().size() == 1);
+        assertTrue(consumer.assignment().containsAll(Arrays.asList(partition1)));
+
+        // respond to the outstanding fetch so that we have data available on the next poll
+        client.respondFrom(fetchResponse(partition1, 0, 0), node);
+        client.poll(0, time.milliseconds());
+
+        // the next poll should return the completed fetch
+        consumer.poll(0);
+
+        final AtomicBoolean commit1Received = new AtomicBoolean(false);
+        client.prepareResponseFrom(new MockClient.RequestMatcher() {
+            @Override
+            public boolean matches(ClientRequest request) {
+                OffsetCommitRequest commitRequest = new OffsetCommitRequest(request.request().body());
+                OffsetCommitRequest.PartitionData partitionData = commitRequest.offsetData().get(partition1);
+                // 1 record has been returned to the consumer, so the committed offset should be 1
+                if (partitionData.offset == 0) {
+                    commit1Received.set(true);
+                    return true;
+                }
+                return false;
+            }
+        }, offsetCommitResponse(Collections.singletonMap(partition1, Errors.NONE.code())), coordinator);
+
+        // subscription change
+        consumer.subscribe(Arrays.asList(topic2), new ConsumerRebalanceListener() {
+            @Override
+            public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+            }
+
+            @Override
+            public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+                // set initial position so we don't need a lookup
+                for (TopicPartition partition : partitions)
+                    consumer.seek(partition, 0);
+            }
+        });
+
+        assertTrue(consumer.assignment().size() == 1);
+        assertTrue(consumer.assignment().containsAll(Arrays.asList(partition1)));
+
+        client.prepareResponseFrom(syncGroupResponse(Arrays.asList(partition2), Errors.NONE.code()), coordinator);
+
+        consumer.poll(0);
```

Comment:
@hachikuji I am trying to write a few unit tests for KIP-70, but am running into some issues here. In this one, the subscription is set to `topic1` and then to `topic2`; but it seems that at this point I am not mocking some responses. Any idea? Thanks.